Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Pass/Specific/SwapChainPass.h

```diff
@@ -52,22 +52,38 @@ namespace AZ
         protected:
             // Pass behavior overrides
             void BuildInternal() override final;
+            void CreateChildPassesInternal() override final;
             void FrameBeginInternal(FramePrepareParams params) override final;
             
             // WindowNotificationBus::Handler overrides ...
+            // The m_pipelinOutputAttachment need to be recreated when render resolution changed
+            void OnResolutionChanged(uint32_t width, uint32_t height) override;
+            // Swapchain may get resized when window is resized
             void OnWindowResized(uint32_t width, uint32_t height) override;
 
         private:
             // Sets up a swap chain PassAttachment using the swap chain id from the window context 
             void SetupSwapChainAttachment();
 
+            // Create a copy pass to copy pipeline output to swapchain
+            // It's needed if device doesn't support swapchain scaling
+            void CreateCopyPass();
+
             // The WindowContext that owns the SwapChain this pass renders to
             const WindowContext* m_windowContext = nullptr;
 
             // The SwapChain used when rendering this pass
             Ptr<PassAttachment> m_swapChainAttachment;
 
-            RHI::SwapChainDimensions m_swapChainDimensions;
+            // The intermediate attachment used for render pipeline's output
+            Ptr<PassAttachment> m_pipelinOutputAttachment;
+
+            // The pass which copies pipeline output to swapchain 
+            Ptr<Pass> m_copyOutputPass;
```

Comment:
Can you explain a bit about this copy. Aren't we already writing to an intermediate RT and then doing a copy to swapchain so why do we need a second copy?

Reply:
This copy pass implementation is expanding the SwapchainPass the functionality to giving the render pipeline an output attachment other than the swapchain attachment. 
It works with any render pipeline pass template. The intermediate RT of MainPipeline is a choice of the MainPipeline. MainPipeline may choose to remove that RT by changing its pass template. And it won't effect the SwapchainPass's ability of having a different resolution for render pipeline output.